### DISCUSSION TITLE
How can I reduce the steps for upscale when a custom steps count is set ?

### DISCUSSION THREAD

**Discussion by LeFanch** [lllyasviel/Fooocus#3516] [Q&A] [answered]
Hi, I like to use the upscale(x1.5) when I generate images from prompt.
When I use the speed or quality preset, the upscale process uses less steps (26 for quality instead of 60, and 18 dor speed).
But when I use a custom step number in advanced mode, the count of steps used for upscale is the same, and the upscaled image looks not good (and it takes more time).
Is there a way to force the steps number for upscale operation too ?
Thank you.

**Comment by kalle07** [comment, marked as answer]
If I am correct, quality and speed are treated differently ... 
if you with python you can create your own sets

**Comment by LeFanch** [comment]
I found them in modules/flags.py
The good news : if I create a new set like "Semi-quality" and create its constants/values, it is visible in UI without other modification.
(but it's better to not modify code)
